### PR TITLE
feat(refactor): updating network requests

### PR DESCRIPTION
**Pull request by himayand21**

### CODE REVIEW

**Comment by ankita1010** [comment]
PR could be more descriptive.

File: src/pages/notes/components/NoteEditor.js [hunk 1]
```diff
@@ -19,8 +20,10 @@ export const NoteEditor = (props) => {
     const [optionButton, setOptionButton] = useState(false);
     const [show, setShow] = useState(false);
     const [link, setLink] = useState('');
+    const [fullScreenFlag, setFullScreenFlag] = useState(false);
+    const {addToast} = useToast();
 
-    const {editorState, onChange, readOnly} = props;
+    const {editorState, onChange, readOnly, editorRef} = props;
```
Comment (ankita1010) [inline src/pages/notes/components/NoteEditor.js:26]: onChange is a JSX event listener name, preferably make the name of the function more descriptive. But !important.

File: src/styles/components/interactive.scss [hunk 2]
```diff
@@ -27,6 +27,9 @@
         background-color: transparentize($color: $green, $amount: 0.6);
         cursor: pointer;
     }
+    i {
+        font-size: 13px !important;
```
Comment (ankita1010) [inline src/styles/components/interactive.scss:31]: Not preferable (personal opinion) and !important.

File: src/components/navBar/index.js [hunk 3]
```diff
@@ -6,11 +6,10 @@ export const NavBar = (props) => {
...
         <nav className="navbar-container">
             <div className="navbar">
-                <header
-                    className={`header ${onTitleClick ? 'hoverable' : ''}`}
-                    onClick={onTitleClick ? onTitleClick : () => {}}
-                >
-                    Blackboard
+                <header className="header">
+                    <span onClick={onTitleClick ? onTitleClick : () => {}}>
+                        Blackboard
```
[the remark this answers is not in the document]
Comment (himayand21) [reply src/components/navBar/index.js:11]: NavLink would require a separate onClick and a to prop.
Would complicate and increase code complexity.

File: src/pages/notes/components/NoteEditor.js (hunk 1 above)
Comment (himayand21) [reply src/pages/notes/components/NoteEditor.js:26]: Yes.
onChange is a common name, but it has been widely used here in multiple scenarios.
Updating that would require plenty of work.
Postponing this for now.

File: src/styles/components/interactive.scss (hunk 2 above)
Comment (himayand21) [reply src/styles/components/interactive.scss:31]: This seems extremely !important to me now. So I !will get to it !asap.

File: src/components/navBar/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ankita1010) [reply src/components/navBar/index.js:11]: How about adding the role="link" attribute?
Advise please.

File: src/pages/notes/components/NoteEditor.js (hunk 1 above)
Comment (ankita1010) [reply src/pages/notes/components/NoteEditor.js:26]: Need to improve naming conventions. Names need to be more descriptive now onwards.

**Pull request merged**